public/app/features/datasources/components/DataSourceTestingStatus.tsx

```diff
@@ -103,15 +151,17 @@ export function DataSourceTestingStatus({ testingStatus, exploreUrl, dataSource
         <Alert severity={severity} title={message} data-testid={e2eSelectors.pages.DataSource.alert}>
           {testingStatus?.details && (
             <>
-              {detailsMessage}
+              {detailsMessage ? <>{String(detailsMessage)}</> : null}
               {severity === 'success' ? (
                 <AlertSuccessMessage
                   title={message}
                   exploreUrl={exploreUrl}
                   dataSourceId={dataSource.uid}
                   onDashboardLinkClicked={onDashboardLinkClicked}
                 />
-              ) : null}
+              ) : (
+                <AlertErrorMessage moreDetailsLink={String(moreDetailsLink)} />
```

Comment:
```suggestion
                moreDetailsLink ? <ErrorDetailsLink link={String(moreDetailsLink)} /> : <></>
```

This is nit, but I think this would be more readable. Also I would rename the component to make it clear that this show additional error details link, rather than error message.

Reply:
Also if we want to use `moreDetailsLink` only for errors (which I think we do), I would suggest renaming it to `errorDetailsLink` also in pkg/tsdb/mysql/sqleng/handler_checkhealth.go.